lms/djangoapps/bulk_email/models.py

```diff
@@ -96,7 +110,7 @@ def get_users(self, course_id, user_id=None):
         elif self.target_type == SEND_TO_LEARNERS:
             return use_read_replica_if_available(enrollment_qset.exclude(id__in=staff_instructor_qset))
         elif self.target_type == SEND_TO_COHORT:
-            return User.objects.none()  # TODO: cohorts aren't hooked up, put that logic here
+            return self.cohorttarget.cohort.users.filter(id__in=enrollment_qset)
```

Comment:
This handles the case of " a user existed in the chose cohort, but has since unenrolled from the course". Tested on my devstack to confirm.